### ISSUE TITLE
Support for adding Aspire SQL Server?

### ISSUE THREAD

**Issue by mikeleigh1010** [ardalis/CleanArchitecture#887]
It would be helpful if the project could either document or support swapping out from using sql lite to using an aspire configured instance of SQL server.

- .NET SDK Version: 9

Steps to Reproduce:

1.Add configuration for a sql server reference
2.Update reference from UseSqlLite to AddSqlServerDbContext

**Comment by ardalis** [comment]
Agreed 100%

**Comment by benjaminmishra** [comment]
@ardalis - I can give this one a try. I have some ideas around how this can be done.

**Comment by ardalis** [comment]
All yours.
You might look at this related issue:
https://github.com/ardalis/DotNetDataAccessTour/issues/8

**Comment by ardalis** [comment]
@benjaminmishra any update?

**Comment by benjaminmishra** [comment]
@ardalis yes, figured out that it can be done using preprocessor directive like syntax 
So we can have the following line of code in the InfrastructureServiceExtensions.cs file.

```csharp
#if (UseSqlServer)
  services.AddDbContext<AppDbContext>(options => options.UseSqlServer(connectionString));
#else
  services.AddDbContext<AppDbContext>(options => options.UseSqlite(connectionString));
#endif
```

Question: How do we want to expose this as an option ? I was thinking may be we can add some flags while template creation and let the user choose the database type they want to use based on this flag?

[choice parameter](https://devblogs.microsoft.com/dotnet/how-to-create-your-own-templates-for-dotnet-new/#add-a-choice-from-a-list-of-options)

**Comment by ardalis** [comment]
Yes, exactly. A flag during template creation.

**Comment by mikeleigh1010** [comment]
Might be also worth considering if the project should maintain the recommended approach for running migrations with Aspire 

https://learn.microsoft.com/en-us/dotnet/aspire/database/ef-core-migrations

This approach makes use of having the migrations ran from a worker service, this way the connection string can be passed in, and the service can run to completion I believe,

**Comment by ardalis** [comment]
This is done in the latest version.
